addons/mail/models/mail_thread.py

```diff
@@ -1636,17 +1642,26 @@ def message_parse(self, message, save_original=False):
                 stored_date = datetime.datetime.now()
             msg_dict['date'] = stored_date.strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT)
 
-        parent_message = self._get_parent_message(msg_dict)
-        if parent_message:
-            msg_dict['parent_id'] = parent_message.id
-            msg_dict['is_internal'] = bool(parent_message.subtype_id and parent_message.subtype_id.internal)
+        msg_dict.update(self._message_parse_extract_from_parent(self._get_parent_message(msg_dict)))
 
         msg_dict['is_bounce'] = self._message_parse_is_bounce(message, msg_dict)
         msg_dict.update(self._message_parse_extract_payload(message, msg_dict, save_original=save_original))
         if msg_dict['is_bounce']:
             msg_dict.update(self._message_parse_extract_bounce(message, msg_dict))
         return msg_dict
 
+    def _message_parse_extract_from_parent(self, parent_message):
+        """Derive message values from the parent."""
+        if parent_message:
+            # if the message is sent by odoobot, we can consider it to not be internal
+            parent_is_internal = bool(parent_message.subtype_id and parent_message.subtype_id.internal)
+            parent_sent_by_odoobot = parent_message.author_id == self.env.ref('base.partner_root')
```

Comment:
Technically, should be in another commit so that we split the tracking ordering from the internal flag ? That way we can have two explanations, as those are two different issues actually.

Reply:
Separated both changes. The second on is bigger now too (some changes for front-end consistency)